src/accelerate/state.py

```diff
@@ -797,6 +797,18 @@ def default_device(self) -> torch.device:
         else:
             return torch.device("cpu")
 
+    def __getattr__(self, name: str):
+        try:
+            # First try the normal way
+            return object.__getattribute__(self, name)
```

Comment:
Trying to think this through: When we hit `__getattr__`, we already know the attribute wasn't found, right? So can we ever successfully `return object.__getattribute__(self, name)`? Or can we raise the below error right away?

Reply:
Can confirm that yes we can, `getattr` does not know by this point, it's what is immediately hit when doing `getattr()`. This then is what does the actual getattr check.

Reply:
Hmm, are you sure? I modified the method like this:

```python
    def __getattr__(self, name: str):
        try:
            # First try the normal way
            attr = object.__getattribute__(self, name)
        except AttributeError as e:
            if name in self._known_attrs:
                # If not found, give a more contextualized answer
                raise AttributeError(
                    f"`PartialState` object has no attribute `{name}`. "
                    "This happens if `PartialState._reset_state()` was called and "
                    "an `Accelerator` or `PartialState` was not reinitialized."
                )
            else:
                raise e
        breakpoint()
        return attr
```

and never hit the breakpoint when running `test_state_after_reset` or any other test. Attribute access goes via `__getattribute__` and only calls `__getattr__` when the former fails with an `AttributeError`. Therefore, I don't think this `__getattribute__` can succeed.

I guess one reason to still leave it there is so that we can re-trigger the `AttributeError` and re-raise it instead of directly raising `AttributeError`, though I can't come up with a scenario where this is useful.

Reply:
So this is because the implementation needs to be under `__getattribute__` via `__getattr__`, as you said. Thanks :)